crates/typst-library/src/visualize/gradient.rs

```diff
@@ -1284,25 +1284,29 @@ fn process_stops(stops: &[Spanned<GradientStop>]) -> SourceResult<Vec<(Color, Ra
 
 /// Sample the stops at a given position.
 fn sample_stops(stops: &[(Color, Ratio)], mixing_space: ColorSpace, t: f64) -> Color {
+    assert!(!stops.is_empty(), "Cannot sample from gradient with no stops");
```

Comment:
While somewhat sensible, I don't think we need to assert here, since this condition is alread checked when constructing the gradients, and the code that would panic is in this function anyway.